tbpore/cli.py

```diff
@@ -29,3 +47,214 @@ def handle_parse_result(self, ctx, opts, args):
+    type=int,
+    default=1
+)
+@click.option(
+    "-m",
+    "--mem_mb",
+    help="Memory to use in tools that accept a memory limit",
+    type=int,
+    default=8*1024  # 8 GB
+)
+@click.option(
+    "--cleanup/--no-cleanup",
+    "-d/-D",
+    default=False,
+    show_default=True,
+    help="Remove all temporary files on *successful* completion",
+)
+@click.argument("inputs", type=click.Path(exists=True, path_type=Path), nargs=-1)
+@click.pass_context
+def main(
+    ctx: click.Context,
+    verbose: bool,
+    quiet: bool,
+    outdir: Path,
+    inputs: Tuple[Path, ...],
+    recursive: bool,
+    tmp: Path,
+    name: str,
+    threads: int,
+    mem_mb: int,
+    cleanup: bool,
+):
+    """Mycobacterium tuberculosis genomic analysis from Nanopore sequencing data
+
+    INPUTS: Fastq file(s) and/or a directory containing fastq files. All files will
+    be joined into a single fastq file, so ensure thery're all part of the same
+    sample/isolate.
+    """
+    log_lvl = "INFO"
+    if verbose:
+        log_lvl = "DEBUG"
+    elif quiet:
+        log_lvl = "ERROR"
+    logger.remove()
+    logger.add(sys.stderr, level=log_lvl, format=log_fmt)
+    logger.info(f"Welcome to TBpore version {__version__}")
+
+    outdir.mkdir(exist_ok=True, parents=True)
+    if tmp is None:
+        tmp = outdir / TMP_NAME
+    tmp.mkdir(exist_ok=True, parents=True)
+
+    if not inputs:
+        logger.error("No INPUT files given")
+        ctx.exit(2)
+
+    if not name:
+        name = inputs[0].name.split(".")[0]
+        if not name:
+            name = "input"
+
+        logger.debug(f"No sample name found; using '{name}'")
+
+    infile = tmp / f"{name}.fq.gz"
+    if len(inputs) == 1 and inputs[0].is_file():
+        if not inputs[0].suffix == ".gz":
+            with open(inputs[0], "rb") as f_in:
+                with gzip.open(infile, "wb") as f_out:
+                    shutil.copyfileobj(f_in, f_out)
+        else:
+            shutil.copy2(inputs[0], infile)
+
+    else:
+        logger.info("Searching for fastq files...")
+        fq_files = []
+        for p in inputs:
+            if p.is_file():
+                fq_files.append(p)
+            else:
+                fq_files.extend(find_fastq_files(p, recursive))
+        logger.info(f"Found {len(fq_files)} fastq files. Joining them...")
+        concatenate_fastqs(fq_files, infile)
+
+    # todo: download the reference genomes from https://www.nature.com/articles/srep45258#Sec25 and https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0214088
+    # todo: remove contamination
+    # todo: run mykrobe
+    # todo: align decontam to h37rv
+    # todo: bcftools mpileup/call
+    # todo: filter vcf
+    mykrobe = ExternalTool(
+        tool="mykrobe",
+        input=f"-i {infile}",
+        output=f"-o {tmp}/{name}.mykrobe.json",
+        params=f"predict --ont --force -A --debug --format json --min_proportion_expected_depth 0.20 --sample {name} "
```

Comment:
Don't use the `--ont` flag until after the release of mykrobe v0.11. Use https://github.com/mbhall88/head_to_head_pipeline/blob/d7b30eda3cca5fbaf2dc22dbcacd79f7c3a876c9/analysis/resistance_prediction/workflow/rules/mykrobe.smk#L18-L23

See https://github.com/mbhall88/tbpore/issues/2#issue-1196635228

Reply:
I think adding a CLI param for whether to use `-A` is wise. I think it should be off by default. It tells mykrobe to output details for all (~66k) mutations. As such, the JSON file is ~92MB. Also don't think we need `--debug`

Reply:
Done in https://github.com/mbhall88/tbpore/pull/8/commits/aeb1de49fdb00ccef3955f85c75ea5e9ce90b9dc and https://github.com/mbhall88/tbpore/pull/8/commits/b6016a5ff925ef1f64531bf0ff283787dbfe804e